src/client.js

```diff
@@ -36,17 +36,60 @@ function initSocket() {
 
 global.socket = initSocket();
 
-function createElement(Component, props) {
-  if (Component.fetchData) {
-    Component.fetchData(store.getState, store.dispatch,
-                        props.location, props.params);
+const history = useScroll(createHistory)({routes: createRoutes(routes)});
+syncReduxAndRouter(history, store);
+
+let lastMatchedLocBefore;
+let lastMatchedLocAfter;
+
+history.listenBefore((location, callback) => {
+  const loc = location.pathname + location.search + location.hash;
+  if (lastMatchedLocBefore === loc) {
+    return callback();
   }
-  return React.createElement(Component, props);
-}
+
+  match({routes: routes, location: loc}, (err, redirectLocation, nextState) => {
+    if (!err && nextState) {
+      fetchData(nextState.components, store.getState, store.dispatch,
+        location, nextState.params)
+        .then(() => {
+          lastMatchedLocBefore = loc;
+          callback();
+        })
+        .catch(err2 => {
+          console.error(err2, 'Error while fetching data');
+          callback();
+        });
+    } else {
+      console.log('Location "%s" did not match any routes (listenBefore)', loc);
+      callback();
+    }
+  });
+});
+
+history.listen((location) => {
+  const loc = location.pathname + location.search + location.hash;
+  if (lastMatchedLocAfter === loc) {
+    return;
+  }
+
+  match({routes: routes, location: loc}, (err, redirectLocation, nextState) => {
```

Comment:
Same here, in fact for this case I think you just need to use the `createElement` method that was in the previous code and remove this `history.listen` subscriber entirely.